### ISSUE TITLE
LDAP requires user to bind; support comparing userPassword attribute

### ISSUE THREAD

**Issue by bradjones1** [rancher/rancher#17854]
**What kind of request is this (question/bug/enhancement/feature request):**

Enhancement

**Steps to reproduce (least amount of steps as possible):**

1. Set up an OpenLDAP implementation with default values per the Rancher 2.x documentation. Provide a service account for looking up user data which has permission to bind to the DN.
1. Attempt to log in with a user who does not have bind permissions. The request fails because at https://github.com/rancher/rancher/blob/a883cfbcf1d77e2d83656d2c13c943fb02c6a952/pkg/auth/providers/ldap/ldap_client.go#L70-L73 the supplied username (since converted to a DN) and password are used to bind and search for its details. If this bind operation fails, so does the login.

**Result:**

The LDAP user with a valid username/password are prohibited from logging in, as they lack permission to bind to the LDAP directory.

**Other details that may be helpful:**

This may not even be possible; it seems like search-then-bind is the new de-facto standard for username/password authentication against an LDAP server.

Desired result: The user would be logged in through means that do not require bind permission, such as a password compare function. I apologize I'm not an LDAP expert, but it appears many systems support this, in line with https://tools.ietf.org/id/draft-stroeder-hashed-userpassword-values-01.html#rfc.section.3

GitLab offers LDAP login that does not appear to require bind; we use JumpCloud and it appears there is a `userPassword` property available for comparison.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):
- Installation option (single install/HA):

Rancher 2.1.6 HA

**Comment by AdrienKuhn** [comment]
Looks like we have the exact same issue with Rancher 2.2-rc3 and we aren't able to use LDAP authentication. 

We are using JumpCloud too.

**Comment by sra** [comment]
This is the same as  #25165 and is also giving us trouble as JumpCloud directory users. Can get it working using the OpenLDAP connector if you give each affected user bind privs, but that is not desirable for us.

**Comment by ricardojlrufino** [comment]
A have same problem on OpenLdap.

**Comment by bradjones1** [comment]
not stale.

**Comment by sra** [comment]
Yes, please consider fixing this - rancher is the only LDAP integration out of dozens that requires this and the security people hate it.

**Comment by bradjones1** [comment]
Not stale.

**Comment by hshimizu** [comment]
Not stale

**Comment by digiserg** [comment]
Not stale

**Comment by digiserg** [comment]
Not stale, still required

**Comment by digiserg** [comment]
Not stale, it needs fixing
